modules/protocol/protocol.service.ts

```diff
@@ -50,13 +53,28 @@ export class ProtocolService {
             return parseFloat(pool?.dynamicData?.swapFee || '0') * swap.valueUSD;
         });
 
+        const userStakeSyncBlock = await prisma.prismaUserBalanceSyncStatus.findUnique({
+            where: { type: PrismaUserBalanceType.STAKED },
+        });
+
+        const userWalletSyncBlock = await prisma.prismaUserBalanceSyncStatus.findUnique({
+            where: { type: PrismaUserBalanceType.WALLET },
+        });
+
+        const poolSyncBlock = await prisma.prismaLastBlockSynced.findUnique({
+            where: { category: PrismaLastBlockSyncedCategory.POOLS },
+        });
+
```

Comment:
i would prefer this in its own query, since this is cached, so its not up to date and its queried by the frontend regularly which is not showing the data, so its triggering those db queries even though its not used.

Guess you should also be able to get the 3 in a single `findMany` query

Reply:
Refactored it into its own, non-cached query.

I like the approach with three simple DB calls better than looping and IFing over a result array. Not sure if you had something else in mind.